### PR TITLE
chore: use @react-native-community/eslint-config

### PR DESCRIPTION
**Pull request by dratwas**
Summary:
---------

I changed @callstack/eslint-config to @react-native-community/eslint-config and fixed all lint issues.

Test Plan:
----------

Green CI

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,39 +11,41 @@
...
     }
   },
   "prettier": {
-    "proseWrap": "never",
     "singleQuote": true,
-    "trailingComma": "es5"
+    "trailingComma": "all",
+    "bracketSpacing": false,
+    "jsxBracketSameLine": true,
+    "parser": "flow"
```
Comment (thymikee) [inline package.json]: Do we need prettier config at all? Isn't it taken from the eslint config?

File: package.json [hunk 2]
```diff
@@ -11,39 +11,41 @@
...
   },
   "eslintConfig": {
-    "extends": "@callstack",
-    "rules": {
-      "global-require": 0,
-      "no-console": 0
+    "extends": "@react-native-community",
+    "env": {
+      "es6": true,
+      "jest": true,
```
[the remark this answers is not in the document]
Comment (thymikee) [reply package.json]: Ok, that's not super important for now

**Comment by thymikee** [comment]
Thank you so much for making this happen @dratwas! ❤️

**Comment by cpojer** [comment]
This is really fantastic. Thank you so so much for doing this!!

**Pull request merged**